packages/http-client-java/generator/http-client-generator-clientcore-test/src/main/java/encode/array/CommaDelimitedArrayProperty.java

```diff
@@ -69,7 +75,10 @@ public static CommaDelimitedArrayProperty fromJson(JsonReader jsonReader) throws
                 reader.nextToken();
 
                 if ("value".equals(fieldName)) {
-                    value = reader.readArray(reader1 -> reader1.getString());
+                    String valueEncodedAsString = reader.getString();
+                    value = valueEncodedAsString == null
+                        ? null
+                        : new LinkedList<>(Arrays.asList(valueEncodedAsString.split(",")));
```

Comment:
We use `LinkedList` for the array, as this is what used in azure-core on initialization for array.

The list created by `Arrays.asList` cannot be modified. User may do e.g. `model.getValues().add(...)` to modify this list.

Reply:
Do we consider making `value` empty array instead of `null`? 
For serialization/deserialization, there seems no difference, but will avoid NPE if user ever does `model.getValues().add()`.

Reply:
Fixed and added some UT to verify the corner cases
https://github.com/microsoft/typespec/pull/9218/files#diff-6222c97fb639b4106f314a395005e7a470d4d0ef262a6d1f78f7797d32c4b8a5